File: 222.count-complete-tree-nodes.java
/*
 * @lc app=leetcode id=222 lang=java
 *
 * [222] Count Complete Tree Nodes
 */

// @lc code=start
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int countNodes(TreeNode root) {
        return findNull(root) - 1;
    }
    public int findNull(TreeNode root) {
        if (root == null) {
        return 1;
        }
        int left = findNull(root.left);
        int right = findNull(root.right);
        return left + right;
    }
}
// @lc code=end

File: 222.count-complete-tree-nodes.py
#
# @lc app=leetcode id=222 lang=python
#
# [222] Count Complete Tree Nodes
#

# @lc code=start
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def countNodes(self, root):
        """
        :type root: Optional[TreeNode]
        :rtype: int
        """
        return self.findNull(root) - 1
    
    def findNull(self, root):
        if root == None:
            return 1
        left = self.findNull(root.left)
        right = self.findNull(root.right)
        return left + right
        
# @lc code=end

